pkgs/misc/tmux-plugins/default.nix

```
@@ -791,6 +791,44 @@ in rec {
     };
   };
 
+  tmux-sessionx = mkTmuxPlugin {
+    pluginName = "sessionx";
+    version = "0-unstable-2024-09-22";
+    src = fetchFromGitHub {
+      owner = "omerxx";
+      repo = "tmux-sessionx";
+      rev = "508359b8a6e2e242a9270292160624406be3bbca";
+      hash = "sha256-nbzn3qxMGRzxFnLBVrjqGl09++9YOK4QrLoYiHUS9jY=";
+    };
+    nativeBuildInputs = [ pkgs.makeWrapper ];
+    postPatch = ''
+      substituteInPlace sessionx.tmux \
+        --replace "\$CURRENT_DIR/scripts/sessionx.sh" "$out/share/tmux-plugins/sessionx/scripts/sessionx.sh"
+      substituteInPlace scripts/sessionx.sh \
+        --replace "/tmux-sessionx/scripts/preview.sh" "$out/share/tmux-plugins/sessionx/scripts/preview.sh"
+      substituteInPlace scripts/sessionx.sh \
+        --replace "/tmux-sessionx/scripts/reload_sessions.sh" "$out/share/tmux-plugins/sessionx/scripts/reload_sessions.sh"
```

```suggestion
      substituteInPlace sessionx.tmux \
        --replace-fail "\$CURRENT_DIR/scripts/sessionx.sh" "$out/share/tmux-plugins/sessionx/scripts/sessionx.sh"
      substituteInPlace scripts/sessionx.sh \
        --replace-fail "/tmux-sessionx/scripts/preview.sh" "$out/share/tmux-plugins/sessionx/scripts/preview.sh"
      substituteInPlace scripts/sessionx.sh \
        --replace-fail "/tmux-sessionx/scripts/reload_sessions.sh" "$out/share/tmux-plugins/sessionx/scripts/reload_sessions.sh"
```
`--replace` is deprecated. Using `-fail` ensures the build will fail if the text to be replaced is no longer present.